server/channel_state.go

```diff
@@ -165,8 +166,22 @@ func (us *userState) getClientState() UserStateClient {
 }
 
 func (cs *callState) getHostID(botID string) string {
-	var hostID string
+	if cs.HostLocked {
+		return cs.HostID
+	}
+
+	// if current host isn't in the call anymore, remove
+	if _, ok := cs.Users[cs.HostID]; !ok {
+		cs.HostID = ""
+	}
 
+	// don't change hosts if we have one already
+	if cs.HostID != "" {
+		return cs.HostID
+	}
```

Comment:
I admit I find this check a little counter intuitive. I wonder if it would be clearer to return early if the host is already present and avoid the side effect altogether.

Reply:
I agree it's a little counter-intuitive, but I'm not sure how to return earlier -- we have to do the two actions above before checking if we have a host...

Reply:
I was thinking of simply doing this:

```go
	if _, ok := cs.Users[cs.HostID]; ok {
                return cs.HostID
	}
```

And otherwise proceed with the usual "longest standing participant" check. We likely just need to move the `getHostID` call after we set the user in `addUserSession` but it should work.

Reply:
Oh, I see what you mean now -- combine the two checks.

Reply:
@cpoile Not sure it will work as expected unless you also do the latter change in `addUserSession`.

Reply:
Sorry, not sure what you mean by "latter change"

Reply:
@cpoile We need to change the order or the state we act upon won't be the expected one since we add the user to the map after we call `getHostID()`.

https://github.com/mattermost/mattermost-plugin-calls/blob/c3513b7a7458832d443fd991a121cbeedb294265/server/session.go#L133C1-L138

Reply:
And when I was testing, I wasn't looking at the host until I connected two people... which worked. Hah. 😮‍💨